tests/unit/utils/event_test.py

```diff
@@ -294,6 +294,19 @@ def test_event_many_backlog(self):
                 evt = me.get_event(tag='testevents')
                 self.assertGotEvent(evt, {'data': '{0}'.format(i)}, 'Event {0}'.format(i))
 
+    # Test the fire_master function. As it wraps the underlying fire_event,
+    # we don't need to perform extensive testing.
+    def test_send_master_event(self):
+        """"Tests that sending an event through fire_master generates expected event"""
+        with eventpublisher_process():
+            me = event.SaltEvent("minion", SOCK_DIR)
+            data = {"data": "foo1"}
+            me.fire_master(data, "test_master")
+
+            evt = me.get_event(tag="fire_master")
+            self.assertGotEvent(evt, {"data": data, "tag": "test_master", "events": None, "pretag": None})
+
+
```

Comment:
@cachedout Sorry for the delay; I thought it had, happy to revisit the test patch.

I didn't like the approach at all; it's taking magic knowledge from another section of the code instead of abstracting the underlying idea of how do we message from the minion to the master in a sane manner.